File: Java/780.java
__________________________________________________________________________________________________
Runtime: 0 ms
Memory Usage: 33.4 MB
class Solution {
    public boolean reachingPoints(int sx, int sy, int tx, int ty) {
        if (sx == tx && sy == ty) {
            return true;
        } else if (tx == ty || sx > tx || sy > ty) { 
            return false;
        } else if (tx > ty) {
            int subtract = Math.max(1, (tx - sx)/ty);
            return reachingPoints(sx, sy, tx - subtract * ty, ty);                 
        } else {
            int subtract = Math.max(1, (ty - sy)/tx);
            return reachingPoints(sx, sy, tx, ty - subtract * tx);
        }
    }
}
__________________________________________________________________________________________________
sample 31820 kb submission
class Solution {
    public boolean reachingPoints(int sx, int sy, int tx, int ty) {
        while (tx >= sx && ty >= sy) {
            if (tx == sx && ty == sy) return true;
            if (tx >= ty) {
                if (ty == sy) return (tx - sx) % ty == 0;
                tx = tx % ty;
            } else {
                if (tx == sx) return (ty - sy) % tx == 0;
                ty = ty % tx;
            }
        }
        if (tx == sx && ty == sy) return true;
        return false;
    }
}
__________________________________________________________________________________________________

File: Python3/780.py
__________________________________________________________________________________________________
sample 24 ms submission
class Solution:
    def reachingPoints(self, sx: int, sy: int, tx: int, ty: int) -> bool:
        if tx > ty:
            sx, sy, tx, ty = sy, sx, ty, tx
            
        while sx < tx:
            sx, sy, tx, ty = sy, sx, (ty-tx) % tx, tx
        
        return sx == tx and ty >= sy and (ty-sy) % sx == 0
__________________________________________________________________________________________________
sample 28 ms submission
class Solution:
    def reachingPoints(self, sx: int, sy: int, tx: int, ty: int) -> bool:
        while sx < tx and sy < ty:
            tx, ty = tx % ty, ty % tx
        return sx == tx and sy <= ty and (ty - sy) % sx == 0 or \
               sy == ty and sx <= tx and (tx - sx) % sy == 0
__________________________________________________________________________________________________
